src/python/WMComponent/AgentStatusWatcher/DrainStatusPoller.py

```diff
@@ -54,3 +68,46 @@ def getDrainInfo(cls):
         Return drainStats class variable
         """
         return cls.drainStats
+
+    def checkSpeedDrainThresholds(self, speedDrainConfig):
+        """
+
+        """
+        validKeys = ['CondorPriority', 'NoJobRetries', 'EnableAllSites']
+        updateConfig = False
+        # get condor jobs
+        jobs = self.condorAPI.getCondorJobs("", [])
+        if jobs is None:
+            logging.warning("There was an error querying the schedd.  Not checking speed drain thresholds.")
+            return
+
+        for k, v in speedDrainConfig.items():
+            # make sure keys are valid
+            if k in validKeys and isinstance(v['Threshold'], int) and isinstance(v['Enabled'], bool):
+                # and the drain config change isn't active
+                if not v['Enabled']:
+                    logging.info("Checking threshold for %s", k)
+                    # if number of condor jobs is below the threshold, activate speed drain config
+                    if len(jobs) < v['Threshold']:
+                        updateConfig = True
+                        if k == 'CondorPriority':
+                            if self.condorAPI.editCondorJobs("JobStatus=?=1 && (CMS_JobType =?= \"Production\" || CMS_JobType =?= \"Processing\")", "JobPrio", "999999"):
+                                logging.info("Threshold hit for %s, Updated condor priority to 999999 for all jobs.", k)
+                                speedDrainConfig[k]['Enabled'] = True
+                        elif k == 'NoJobRetries':
+                            if self.reqAuxDB.updateMaxRetries(self.config.Agent.hostName, 0):
+                                logging.info("Threshold hit for %s. Updated max retries to 0.", k)
+                                speedDrainConfig[k]['Enabled'] = True
+                        elif k == 'EnableAllSites':
+                            logging.info("Updating agent to submit to all sites.")
+                            # setting this value to True makes JobSubmitterPoller ignore site status
+                            speedDrainConfig[k]['Enabled'] = True
+                else:
+                    logging.info("Speed drain configuration is active for %s", k)
```

Comment:
This is not correct for `CondorPriority`, which is supposed to go through a continuous update (when it's enabled).